contracts/sophia.md

```diff
@@ -814,9 +840,16 @@ private public rec stateful switch true type record datatype
 - `QCon = (Con\.)+Con` qualified constructor
 - `TVar = 'Id` type variable (e.g `'a`, `'b`)
 - `Int = [0-9]+|0x[0-9A-Fa-f]+` integer literal
-- `Hash = #[0-9A-Fa-f]+` hash literal
+- `Bytes = #[0-9A-Fa-f]+` byte array literal
```

Comment:
Type `bytes(n)` is defined above as "A byte array with `n` bytes". This grammar allows for odd number of hexadecimal digits.

Is an odd number of hexadecimal legal?
If yes, is it equivalent to same literal prepended by one zero?

Reply:
Is any canonical representation enforced by compiler e.g. no preceding zeros?

Reply:
An odd number of hex-digits means there is an implicit extra `0`...

Nothing is enforced by the compiler, leading zeros have a meaning.